wire-ios-sync-engine/Source/SessionManager/SessionManager+AnalyticsUseCases.swift

```diff
@@ -0,0 +1,56 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+extension SessionManager {
+
+    enum SessionManagerError: Error {
```

Comment:
The name of this error suggests it's for general errors within the session manager, but it's all to do with analytics. I suggest a more accurate name.

Reply:
You're absolutely right the name doesn't reflect the for those types of errors. Updated it here: https://github.com/wireapp/wire-ios/commit/5690c91fecccb9747ee0e35b7fd6f2ee9012d2de